core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -585,7 +621,12 @@ private void initializeWriter(StoreConfiguration cfg, CacheWriter writer, Initia
       if (writer != null) {
          if (writer instanceof DelegatingCacheWriter)
```

Comment:
Yes I agree; reading the SingletonCacheWriter it seems to apply it only works with shared cache stores as I would think.  Unfortunately our documentation and verification (`PersistenceConfigurationBuilder`) have it the other way.